### DISCUSSION TITLE
O Google alcança e indexa frameworks JavaScript? Um experimento de SEO com JS

### DISCUSSION THREAD

**Discussion by fititnt** [frontendbr/forum#579] [Dicas]
[Can Google properly crawl and index JavaScript frameworks? JavaScript SEO experiment.
https://goralewicz.com/blog/javascript-seo-experiment/](https://goralewicz.com/blog/javascript-seo-experiment/)

Repositório com os códigos: https://github.com/kamilgrymuza/jsseo

---
TL;DR imagens:

> What is really interesting is Googlebot is fully processing some of the major JavaScript frameworks, but isn’t doing very well with Angular 1 and Angular 2, when both of those frameworks are built by Google.

![image](https://cloud.githubusercontent.com/assets/812299/26238856/36f38804-3c51-11e7-8bf9-1408016d3090.png)

> When designing this experiment, I went through a lot of Google patents related to crawling efficiency. I was expecting that Google may try to optimize processing JavaScript, as crawling JavaScript requires more resources compared to plain HTML and CSS.
> 
> External JavaScript files were hosted on the same server and within the same domain.

![image](https://cloud.githubusercontent.com/assets/812299/26238903/67d607ee-3c51-11e7-9279-0fed05377caf.png)

> Experiment Results - React - Inline vs. External

![image](https://cloud.githubusercontent.com/assets/812299/26238914/744e256a-3c51-11e7-9d61-fd124e931315.png)

> Experiment Results - Plain JavaScript - Internal vs. External

![image](https://cloud.githubusercontent.com/assets/812299/26238921/8055a6f8-3c51-11e7-9e9b-0376f07d12ba.png)

> Experiment Results - Angular JS 1 and 2 - Inline, vs External vs. Bundled

![image](https://cloud.githubusercontent.com/assets/812299/26238941/92bb4ad2-3c51-11e7-9dcc-289165d00e4c.png)

----

Repositório com os códigos: https://github.com/kamilgrymuza/jsseo

**Comment by nauseos** [comment]
Hi, I'm just studying Angular yet, but as far as I could understand, it wouldn't happen when using server rendering.

**Comment by felipefialho** [comment]
@nauseos Pode usar ptbr para comentar aqui :)

**Comment by nauseos** [comment]
@LFeh Kk.. Nem reparei! Flw
